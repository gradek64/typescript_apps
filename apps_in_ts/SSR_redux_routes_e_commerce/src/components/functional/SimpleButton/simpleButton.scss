.simple-button {
  width: 100%;
  padding: 0.5rem 1rem;
  border-radius: 5rem;
  background-color: lightblue;
  cursor: pointer;
}
