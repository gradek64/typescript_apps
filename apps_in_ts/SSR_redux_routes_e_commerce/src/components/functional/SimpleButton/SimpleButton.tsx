import React from 'react';
// custom
import './simpleButton.scss';

interface InterfaceProps {
  label: string,
  eventHandler: (e: React.MouseEvent) => void,
}
const SimpleButton = ({ label, eventHandler }: InterfaceProps) =>
  (<input className="simple-button" type="button" onClick={eventHandler} value={label} />);
export default SimpleButton;
