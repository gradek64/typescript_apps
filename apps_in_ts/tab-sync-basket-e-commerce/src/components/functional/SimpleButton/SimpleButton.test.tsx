import React from 'react';
import { render, fireEvent } from '@testing-library/react'

import SimpleButton from './SimpleButton';
const props = {
  label: 'add',
  eventHandler: jest.fn(),
}
describe('render', () => {
  it('renders correctly with promotion', () => {
    const { asFragment } = render(<SimpleButton {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
describe('events', () => {
  it('executes event handler on click', () => {
    const { eventHandler } = props
    const { container } = render(<SimpleButton {...props} />);
    const inputButton = container.querySelector('.simple-button')
    fireEvent.click(inputButton as Element)
    expect(eventHandler).toBeCalled()
  })
})
