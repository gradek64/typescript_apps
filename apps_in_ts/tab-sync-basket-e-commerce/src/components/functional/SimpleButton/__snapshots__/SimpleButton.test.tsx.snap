// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly with promotion 1`] = `
<DocumentFragment>
  <input
    class="simple-button"
    type="button"
    value="add"
  />
</DocumentFragment>
`;
